web_button_invisible/models/hide_button_rule.py

```
@@ -0,0 +1,72 @@
+# Copyright 2023 ooops404
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+
+from odoo import _, api, exceptions, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class HideButtonRule(models.Model):
+    _name = "hide.button.rule"
+    _description = "Rule to hide a button"
+
+    button_name = fields.Char(required=True)
+    model_id = fields.Many2one(
+        "ir.model",
+        ondelete="cascade",
+        index=True,
+    )
+    model_name = fields.Char(related="model_id.model")
+    condition_domain = fields.Char()
+    group_ids = fields.Many2many("res.groups", required=True)
+    # generated technical fields used in form attrs:
+    button_visibility_field_id = fields.Many2one("ir.model.fields", ondelete="cascade")
```

I would add a `action` selection field which by default is `hide`.